packages/react/src/components/FileUploader/FileUploaderDropContainer.js

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, matches } from '../../internal/keyboard';
+import uniqueId from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function FileUploaderDropContainer(props) {
+  const inputRef = useRef();
+  const {
+    accept,
+    className,
+    id,
+    disabled,
+    labelText,
+    multiple,
+    name,
+    onAddFiles,
+    role,
+    tabIndex,
+    ...other
+  } = props;
+  const uid = useRef(uniqueId()).current;
+  const [isActive, setActive] = useState(false);
+  const classes = classNames(`${prefix}--file__drop-container`, {
+    [`${prefix}--file__drop-container--drag-over`]: isActive,
+    [className]: className,
+  });
+  const handleChange = useCallback(
+    evt => {
+      onAddFiles(evt, {
+        addedFiles:
+          evt.type === 'drop'
+            ? [...evt.dataTransfer.files]
+            : [...evt.target.files],
+      });
+    },
+    [onAddFiles]
+  );
+  return (
+    <div
+      className={`${prefix}--file`}
+      onDragOver={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(true);
+        evt.dataTransfer.dropEffect = 'copy';
+      }}
+      onDragLeave={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(false);
+        evt.dataTransfer.dropEffect = 'move';
+      }}
+      onDrop={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(false);
+        handleChange(evt);
+      }}>
+      <label
+        className={`${prefix}--file-browse-btn`}
+        htmlFor={id || uid}
+        role={role || 'button'}
```

Comment:
Given that we're using the button role, do we need to do anything special? Also, should we allow this to be configurable if we are implementing onClick/onKeyDown to match this a11y role?

Reply:
not sure what you mean by doing anything special but `label`s are interactable elements so they support event handlers, and the role is meant to show that the label functions like a button

Reply:
If an element has `role` set to `button` it needs to implement handlers for `onClick` and `onKeyDown` https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Required_JavaScript_Features

In this case, it appeared to be only `onKeyDown`

Reply:
in this case `label` already handles click events because it is associated with an `input` element